src/parse/parse_mat.jl

```diff
@@ -0,0 +1,1667 @@
+# CONTRIB file
+# Originally from PowerModels.jl. Adapted by PowerSystems.jl and by us.
+
+#########################################################################
+#                                                                       #
+# This file provides functions for interfacing with Matpower data files #
+#                                                                       #
+#########################################################################
+
+module ParseMAT
+
+using ..ExaPF: IdxSet
+
+include("matlab.jl")
+include("im_common.jl")
+
+"Parses the matpwer data from either a filename or an IO object"
+function parse_matpower(io::IO)::Dict
+    mp_data = _parse_matpower_string(read(io, String))
+    #pm_data = _matpower_to_powermodels!(mp_data)
+    return mp_data
+end
+
+function parse_mat(file::String; kwargs...)::Dict
+    mp_data = open(file) do io
+        parse_matpower(io; kwargs...)
+    end
+    return mp_data
+end
+
+function mat_to_exapf(data_mat)
+    
+    # create new data structure
+    data = Dict{String,Array}()
+    
+    # baseMVA
+    data["baseMVA"] = [data_mat["baseMVA"]]
+
+    # buses
+    BUS_I, BUS_TYPE, PD, QD, GS, BS, BUS_AREA, VM, VA, BASE_KV, ZONE, VMAX, VMIN,
+    LAM_P, LAM_Q, MU_VMAX, MU_VMIN = IdxSet.idx_bus()
+    
+    nbus = length(data_mat["bus"])
+    bus_array = zeros(nbus, 17)
+
+    for (i, bus) in enumerate(data_mat["bus"])
+        busn = i # we need an external2internal function
```

Comment:
Yes! Today, I learned that almost all matpower instances have their specific IDs for buses. For instance, to import instances from `pglib-opf`, I have to manually build the `external2internal` function when building the adjacency matrix here: https://github.com/exanauts/ExaPF.jl/blob/adrian/matpower/src/powersystem.jl#L378L379
and I added the following lines to import the pglib instances:
```julia
        # build connection matrices
    f = branch[:, F_BUS]                           # list of "from" buses
    t = branch[:, T_BUS]                            # list of "to" buses

    # Convert to valid ordered indexing for matrices
    bus_indexes = convert.(Int, bus[:, BUS_I])
    # in bus_indexes, index shall be unique
    f_i = [findfirst(isequal(el), bus_indexes) for el in f]
    t_i = [findfirst(isequal(el), bus_indexes) for el in t]
```

I do not know if this is the most robust solution, though